src/lib.rs

```diff
@@ -1,3 +1,196 @@
+mod origin {
```

Comment:
what's the reasoning behind having all these module blocks?

Reply:
good point. i think i started with a copy-paste example and just kept going. i should fix this